### PR TITLE
Transfer of qc2udf_artsample.py to cg_lims

### PR DESCRIPTION
**Pull request by RasmusBurge-CG**
### Added
Added qc_to_sample.py which is a new version of qc2udf_art2samp.py, see [#179](https://github.com/Clinical-Genomics/cg_lims/issues/179)

**Steps to consider while deploying**
- Configuration changes:
- Documentation updates:
- Inform users by email:

### Review:
- [ ] Code approved by
- [ ] Tests executed on stage by:  (Document the test done with screen shots and description.)
- [ ] "Merge and deploy" approved by

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: cg_lims/EPPs/udf/copy/qc_to_sample.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+from cg_lims import options
+from cg_lims.exceptions import LimsError, MissingUDFsError
+from cg_lims.get.artifacts import get_artifacts
+
+
+def artifacts_to_sample(
+    artifacts: list, sample_qc_udf: str = None
+) -> None:
+
+    """Function to copy artifact qc flag to sample level.
```
Comment (mayabrandi) [inline cg_lims/EPPs/udf/copy/qc_to_sample.py:17]:
```suggestion
    """Function to copy artifact qc flag to sample UDF.
```

File: cg_lims/EPPs/udf/copy/qc_to_sample.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    is a list with many samples. Otherwiese a list with only one sample.
+
+    Arguments:
+        artifacts: list of artifacts to copy from
+        sample_qc_udf: sample qc udf to set based on artifact qc_flag"""
+
+    failed_udfs = 0
+    passed_udfs = 0
+    for art in artifacts:
+        udf = art.qc_flag
```
Comment (mayabrandi) [inline cg_lims/EPPs/udf/copy/qc_to_sample.py:30]:
```suggestion
```

File: cg_lims/EPPs/udf/copy/qc_to_sample.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    to all samples related to the artifact. If a art is a pool, art.samples
+    is a list with many samples. Otherwiese a list with only one sample.
+
+    Arguments:
+        artifacts: list of artifacts to copy from
+        sample_qc_udf: sample qc udf to set based on artifact qc_flag"""
+
+    failed_udfs = 0
+    passed_udfs = 0
+    for art in artifacts:
```
Comment (mayabrandi) [inline cg_lims/EPPs/udf/copy/qc_to_sample.py:29]:
```suggestion
    for artifact in artifacts:
```

replace art with artifact at all places below

File: cg_lims/EPPs/udf/copy/qc_to_sample.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        udf = art.qc_flag
+        if udf is not None:
```
Comment (mayabrandi) [inline cg_lims/EPPs/udf/copy/qc_to_sample.py:31]:
```suggestion
        if artifact.qc_flag is not None:
```

File: cg_lims/EPPs/udf/copy/qc_to_sample.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        )            
+
+@click.command()
+@options.sample_qc_udf(help="Name of sample QC udf to set.")
+@options.input(
+    help="Use this flag if you want copy udfs from input artifacts. Defaulte is output artifacts."
+)
+@click.pass_context
+def qc_to_sample(ctx, input,sample_qc_udf):
+    """Script to copy artifact QC-flag to sample"""
```
Comment (mayabrandi) [inline cg_lims/EPPs/udf/copy/qc_to_sample.py:53]:
```suggestion
    """Script to copy artifact QC-flag to sample UDF"""
```

**Review by mayabrandi — COMMENTED** [review]
Great Rasmus! You've made your first EPP! 😃 
I had a few minor comments. 
Would be great if you could add some logging to the EPP as well. You can look at[ this EPP](https://github.com/Clinical-Genomics/cg_lims/blob/master/cg_lims/EPPs/udf/calculate/calculate_water_volume_rna.py) eg.

And we need to add a test of course

**Comment by karlnyr** [comment]
🔥

File: cg_lims/EPPs/udf/copy/qc_to_sample.py (hunk 2 above)
Comment (RasmusBurge-CG) [reply cg_lims/EPPs/udf/copy/qc_to_sample.py]: Agree, changed in the new commit

**Comment by AnnaLeinfelt** [comment]
Nice! Remember to fill in issue template with information.

**Pull request merged**